File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/chats/MessageCreationRequest.java
package ru.vitaliy.petrov.server.forms.requests.chats;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class MessageCreationRequest {

    @NotNull
    private final Long addressee;

    @NotNull
    private final Date messageCreationDate;

    @NotNull
    private final Time messageCreationTime;

    @NotNull
    @NotEmpty
    private final String messageText;

    @NotNull
    private final Long chatID;
}

File: 3.client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/chats/MessageCreationRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.chats

import java.sql.Date
import java.sql.Time

data class MessageCreationRequest(
    val addressee: Long,
    val messageCreationDate: Date,
    val messageCreationTime: Time,
    val messageText: String,
    val chatID: Long
)
